### PR TITLE
Registo e Login de Clientes. Povoacao da base de dados

### PR DESCRIPTION
**Pull request by franciscomanuelmatos**
Endpoints de registo e login de clientes.

#7 #8

### CODE REVIEW

File: src/main/java/pt/tqs/g205/security/JwtAuthorizationFilter.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    if (header != null && header.startsWith("Bearer ")) {
+      UsernamePasswordAuthenticationToken auth = getAuthentication(request, header.substring(7));
+      if (auth != null) {
+        SecurityContextHolder.getContext().setAuthentication(auth);
+      }
+    }
+    chain.doFilter(request, response);
+  }
+
+  private UsernamePasswordAuthenticationToken getAuthentication(HttpServletRequest request,
```
Comment (tqs-go-nutri) [inline src/main/java/pt/tqs/g205/security/JwtAuthorizationFilter.java]: Code Smell Code Smell: Remove this unused method parameter "request". ([squid:S1172](https://sonarcloud.io/organizations/bdomingues-github/rules#rule_key=squid%3AS1172))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=pt.tqs.g205%3Arestapi&pullRequest=14&issues=AWOs6QOC_Tl7jhO82Lfm&open=AWOs6QOC_Tl7jhO82Lfm)

[ISSUE_KEY]: AWOs6QOC_Tl7jhO82Lfm

File: src/main/java/pt/tqs/g205/security/JwtAuthorizationFilter.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
```
Comment (tqs-go-nutri) [inline src/main/java/pt/tqs/g205/security/JwtAuthorizationFilter.java]: Code Smell Code Smell: Remove this unused method parameter "request". ([squid:S1172](https://sonarcloud.io/organizations/bdomingues-github/rules#rule_key=squid%3AS1172))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=pt.tqs.g205%3Arestapi&pullRequest=14&issues=AWOs6QOC_Tl7jhO82Lfm&open=AWOs6QOC_Tl7jhO82Lfm)

[ISSUE_KEY]: AWOs6QOC_Tl7jhO82Lfm

Comment (tqs-go-nutri) [inline src/main/java/pt/tqs/g205/security/JwtAuthorizationFilter.java]: Code Smell Code Smell: Remove this unused method parameter "request". ([squid:S1172](https://sonarcloud.io/organizations/bdomingues-github/rules#rule_key=squid%3AS1172))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=pt.tqs.g205%3Arestapi&pullRequest=14&issues=AWOs6QOC_Tl7jhO82Lfm&open=AWOs6QOC_Tl7jhO82Lfm)

[ISSUE_KEY]: AWOs6QOC_Tl7jhO82Lfm

**Review by josepp-moreira — APPROVED** [review]
Looks good

**Pull request merged**